src/main/scala/com/fulcrumgenomics/bam/OverlappingBasesConsensusCaller.scala

```diff
@@ -0,0 +1,250 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2022 Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.bam
+
+import com.fulcrumgenomics.bam.api.{SamRecord, SamSource}
+import com.fulcrumgenomics.commons.collection.SelfClosingIterator
+import com.fulcrumgenomics.commons.util.Logger
+import com.fulcrumgenomics.util.NumericTypes.PhredScore
+
+
+/** Consensus calls the overlapping mapped bases in read pairs.
+  *
+  * This will iterate through the mapped bases that overlap between the read and mate in a read pair.  If the read and
+  * mate agree at a given reference position, then read and mate base will not change and the base quality returned
+  * is controlled by `maxQualOnAgreement`. If they disagree at a given reference position, then the base and quality
+  * returned is controlled by `onlyMaskDisagreements`.
+  *
+  * @param onlyMaskDisagreements if the read and mate bases disagree at a given reference position, true to mask (make
```

Comment:
Ditto here.  `maskDisagreements`

Reply:
Having gone through the code I now understand what this does, and the documentation is _really_ confusing to me.  What you're really saying, I think is:

`If true: any time bases disagree mask them to N with qual=2, and any time bases agree pass the bases and quality scores through unmodified.  If false: when bases disagree set the output base to be the higher quality base and the qual the delta between the quality scores, and when bases agree adjust the base quality based on `maxQualOnAgreement`.

I think this would make a lot more sense to rework the options as two "strategies":

1. `disagreementStrategy`: one of `MaskBoth`, `MaskLowerQual`, `Consensus`
2. `agreementStrategy`: one of `Consensus`, `MaxQual`, `PassThrough`

Reply:
it is done.